File: model/src/main/java/io/quartic/geojson/LineString.java
package io.quartic.geojson;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import io.quartic.common.SweetStyle;
import org.immutables.value.Value;

import java.util.List;

@SweetStyle
@Value.Immutable
@JsonSerialize(as = LineStringImpl.class)
@JsonDeserialize(as = LineStringImpl.class)
public interface LineString extends Geometry {
    List<List<Double>> coordinates();

    @Override
    default <T> T accept(GeometryVisitor<T> geometryVisitor) {
       return geometryVisitor.visit(this);
    }
}

File: common-core/src/main/kotlin/io/quartic/common/geojson/LineString.kt
package io.quartic.common.geojson

import com.fasterxml.jackson.annotation.JsonTypeName

@JsonTypeName("LineString")
data class LineString @JvmOverloads constructor(
        val coordinates: List<List<Double>>,
        override val crs: Map<String, Any>? = null,
        override val bbox: List<Double>? = null
) : Geometry
